### DISCUSSION TITLE
how to use translatable package such as Spatie laravel translatable

### DISCUSSION THREAD

**Discussion by abanghendri** [Z3d0X/filament-fabricator#172] [Q&A]
Hi, is there any way to use [Spatie Laravel-translatable to blocks?](https://spatie.be/docs/laravel-translatable/v6/introduction)
what I already did are:

1. install spatie laravel translatable then install fabricator then add my own field, since spatie translatable use json, I chaneg title and slug to json format and leave blocks since it already in json format
2. I also use my own PageResource because as mentioned above, I use some custom field
3. I can save my page, and query my slug, spatie use default language as described on config file
4. when I visit my db, it looks like this: `{"en":"about"}`
5. by my blocks still look like this: `[{"type":"rich-text","data":{"content":"<p>hallo</p>"}}]` without translation

so, How to achieve that?

**Comment by Voltra** [comment]
Please refer to https://github.com/Z3d0X/filament-fabricator/discussions/186 in which I detailed a way to do it
